packages/nimble-components/src/anchor-menu-item/index.ts

```diff
@@ -0,0 +1,125 @@
+import { attr, observable } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    AnchorOptions,
+    MenuItem as FoundationMenuItem
+} from '@microsoft/fast-foundation';
+import { styles } from './styles';
+import { template } from './template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-anchor-menu-item': AnchorMenuItem;
+    }
+}
+
+/**
+ * Types of anchor menu item column count.
+ * @public
+ */
+export type AnchorMenuItemColumnCount = 0 | 1;
+
+/**
+ * A nimble-styled anchor menu-item
+ *
+ * This must extend FoundationMenuItem (as opposed to AnchorBase), because the FAST menu has logic
+ * for properly indenting items that only works with FAST menu items.
+ */
```

Comment:
At a surface level I don't see this and FAST's intention is generally not to be that tightly coupled. For instance, the menu is using [checks for roles](https://github.com/microsoft/fast/blob/1be91f1b6212a55963cdaaa0cd8af76b4d7091fa/packages/web-components/fast-foundation/src/menu/menu.ts#L278) from what I see, not specific logic "that only works with FAST menu items".

Reply:
Yeah, for the most part that's true, but I ran into [this](https://github.com/microsoft/fast/blob/8327524eaf8175c89c435275974243ab09f3afed/packages/web-components/fast-foundation/src/menu/menu.ts#L252). This code for indenting has since been replaced, but it seems like we currently have to make our type a FoundationMenuItem, or we won't indent properly.

Reply:
We can override [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/hasInstance) behavior as a workaround.

To me (without having dived deep into implementation details) it seems like the ideal approach is to able to leverage AnchorBase instead of menu-item as the base class as it has other unrelated behaviors we don't care about. i.e. leveraging menu-item. We would characterize leveraging MenuItem as the base class as tech debt to resolve after migrating to fast-foundation v3.

If that interpretation seems right then maybe this workaround of manipulating the hasInstance check is worth it to reduce the amount of techdebt being added. If not I think we should discuss it more.